R/calcPastureSuit.R

```diff
@@ -29,88 +29,80 @@ calcPastureSuit <- function(climatetype = "MRI-ESM2-0:ssp126",
   }
 
   # Drivers of managed pastures
-  population <- calcOutput("GridPop", subtype = "all", cellular = TRUE, FiveYear = TRUE, harmonize_until = 2015,
-                           aggregate = FALSE)[, , toupper(substring(x$scenario, first = 0, last = 4))]
-
-  cellPrep   <- calcOutput("LPJmLClimateInput", climatetype  = climatetype,
-                           variable     = "precipitation:monthlySum",
-                           stage        = stage,
-                           lpjmlVersion = lpjml,
-                           aggregate    = FALSE)
+  population <- calcOutput("GridPop",
+    subtype = "all", cellular = TRUE, FiveYear = TRUE, harmonize_until = 2015,
+    aggregate = FALSE
+  )[, , toupper(substring(x$scenario, first = 0, last = 4))]
+
+  cellPrep <- calcOutput("LPJmLClimateInput",
+    climatetype = climatetype,
+    variable = "precipitation:monthlySum",
+    stage = stage,
+    lpjmlVersion = lpjml,
+    aggregate = FALSE
+  )
+
+
+  cellPet <- calcOutput(
+    type = "LPJmL_new", climatetype = climatetype,
+    subtype = "mpet",
+    stage = stage,
+    version = lpjml,
+    aggregate = FALSE
+  )
+
+
+  yearsCellPet <- intersect(getYears(cellPet), findset("time"))
+  yearsCellPrep <- intersect(findset("time"), getYears(cellPrep))
+  years <- intersect(yearsCellPet, yearsCellPrep)
+  cellPrep <- dimSums(cellPrep[, years, ], dim = 3)
+  cellPet <- dimSums(cellPet[, years, ], dim = 3)
 
+  # Cell area calculation
+  land <- calcOutput("LanduseInitialisation", input_magpie = TRUE,
+                     aggregate = FALSE, cellular = TRUE, cells = cells,
+                     years = "y1995", round = 6)
+  landArea <- setYears(dimSums(land, dim = 3), NULL)
 
-  cellPet    <- calcOutput(type = "LPJmL_new", climatetype = climatetype,
-                           subtype   = "mpet",
-                           stage     = stage,
-                           version   = lpjml,
-                           aggregate = FALSE)
-
-  yearsCellPet      <- intersect(getYears(cellPet), findset("time"))
-  yearsCellPrep     <- intersect(findset("time"), getYears(cellPrep))
-  years             <- intersect(yearsCellPet, yearsCellPrep)
-  cellPrep          <- dimSums(cellPrep[, years, ], dim = 3)
-  cellPet           <- dimSums(cellPet[, years, ], dim = 3)
+  cellorder <- population
+  cellorder <- match(dimnames(cellorder)$x.y.iso, dimnames(landArea)$x.y.iso)
 
-  # Cell area calculation
-  landArea <- calcOutput("LandArea", aggregate = FALSE) / 100 # transformed to km^2
+  # Reorder the dimensions of landArea to match Population
+  landArea <- landArea[cellorder, , ]
+  getCells(landArea) <- getCells(population)
```

Comment:
the reordering and renaming also looks dangerous to me. Why is this required? It should not be required

Reply:
Fixed